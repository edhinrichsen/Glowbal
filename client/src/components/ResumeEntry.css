.resume-entry {
  margin-top: 30px;
}

.resume-entry-inline ul {
  padding: 0;
  list-style-type: none;
}
.resume-entry-inline li {
  padding: 10px;
  margin: 5px;
  background-color: var(--background-color-secondary);
  filter: var(--drop-shadow);
  border-radius: 100px;
  /* border: 1px solid; */
  min-width: 50px;
  display: inline-block;
}

.resume-entry-inline .resume-entry-date{
  display: none;
}

.resume-entry-inline .resume-sub-text{
  display: none;
}

.resume-entry-block ul {
  padding: 0;
  text-align: left;
  list-style-type: none;
  margin: 0;
}
.resume-entry-block li {
  padding: 5px;
  margin-bottom: 5px;
  background-color: var(--background-color-secondary);
  filter: var(--drop-shadow);
  border-radius: var(--border-radius-secondary);
  display: grid;
  grid-template-columns: 85px auto;
  /* border: 1px solid; */
  /* min-width: 50px; */
  /* display: inline-block; */
}

.resume-entry-block .resume-entry-date{
  font-family: 'Rubik Mono One', sans-serif;
  font-size: 0.75em;
  text-align: right;
  margin-right: 10px;
  padding-right: 10px;
  overflow-x: hidden;
  margin-top: 4px;
  border-right: 1px solid var(--color-app);
  opacity: 0.6;
  
}

.resume-entry-block .resume-sub-text{
  opacity: 0.6;
}
