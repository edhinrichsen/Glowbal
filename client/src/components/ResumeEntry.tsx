import React from 'react';

import './ResumeEntry.css';

type  ResumeEntryProps = {
    title: string,
    display: string,
    entries: {date?: string, text: string, subText?: string}[]
}

function ResumeEntry(props: ResumeEntryProps) {

    return (
        <div className={`resume-entry ${props.display === "inline" ? "resume-entry-inline" : "resume-entry-block"}`} >
            <h2><strong>{props.title}</strong></h2>
            <ul>
            {props.entries.map( (value, index, array) => {
                    return (
                    <li><div className="resume-entry-date">{value.date}</div><div>{value.text}<div className="resume-sub-text">{value.subText}</div></div></li>
                    )
                })}
            </ul>
        </div>
            
    )
}

export default ResumeEntry;