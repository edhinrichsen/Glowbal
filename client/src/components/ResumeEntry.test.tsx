import React from 'react';
import { shallow } from 'enzyme';
import ResumeEntry from './ResumeEntry';

describe('Testing ResumeEntry', () => {
  it('renders without crashing', () => {
    const component = shallow(
      <ResumeEntry
        title="Experience"
        display="block"
        entries={[
          {
            date: '2020',
            text: 'Internship at a Company',
            subText: 'as a Full Stack Developer',
          },
          {
            date: '2019',
            text: 'Internship at a different Company',
            subText: 'as a Front end Developer',
          },
          { date: '2018-Current', text: 'Part time job' },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
