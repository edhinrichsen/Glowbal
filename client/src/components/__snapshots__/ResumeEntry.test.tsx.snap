// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ResumeEntry renders without crashing 1`] = `
<div
  className="resume-entry"
>
  <Spin
    size="large"
    spinning={true}
    wrapperClassName=""
  />
</div>
`;
